sys/kern/vmem.c

```diff
@@ -76,11 +76,17 @@ typedef struct bt {
 static POOL_DEFINE(P_VMEM, "vmem", sizeof(vmem_t));
 static POOL_DEFINE(P_BT, "vmem boundary tag", sizeof(bt_t));
 static alignas(PAGESIZE) uint8_t P_VMEM_BOOTPAGE[PAGESIZE];
-static alignas(PAGESIZE) uint8_t P_BT_BOOTPAGE[PAGESIZE];
+/* Note: in the future, the amount of static memory for boundary tags should
+ * be reduced by more clever tag allocation technique that always keeps some
+ * number of free tags. For more information, please see bt_alloc and bt_refill
+ * methods in NetBSD's vmem and M_NOGROW flag in Mimiker. */
+#define VMEM_BT_BOOTPAGE_CNT 2
+static alignas(PAGESIZE) uint8_t P_BT_BOOTPAGE[VMEM_BT_BOOTPAGE_CNT][PAGESIZE];
 
 void vmem_bootstrap(void) {
   pool_add_page(P_VMEM, P_VMEM_BOOTPAGE);
-  pool_add_page(P_BT, P_BT_BOOTPAGE);
+  for (int i = 0; i < VMEM_BT_BOOTPAGE_CNT; i++)
+    pool_add_page(P_BT, P_BT_BOOTPAGE[i]);
```

Comment:
Could you actually slightly change `pool_add_page` interface to accept size of second argument aligned to pagesize? I'd like to have:

```
static alignas(PAGESIZE) uint8_t P_BT_BOOTPAGE[PAGESIZE*2];

void vmem_bootstrap(void) {
  ...
  pool_add_page(P_BT, P_BT_BOOTPAGE, sizeof(P_BT_BOOTPAGE));
}
```

Reply:
This will require to modify slightly `add_slab` as well, but that's ok.

Reply:
Should it create one big slab (so we also need to add field `ph_size` to `pool_slab_t`, as it'll needed in `destroy_slab`), or many single-paged slabs?

Reply:
Yes. One big slab is preferred solution.

Reply:
Implemented in https://github.com/cahirwpz/mimiker/pull/693/commits/b42c0feec66a339a33a2bed40373d183962e56fb